In [4]:
import numpy as np

#### Task 4.
``` python
my_array = np.array([[1,2,3,4,5],
                     [6,7,8,9,10],
                     [11,12,13,14,15],
                     [16,17,18,19,20],
                     [21,22,23,24,25]])
```
Перед вами массив my_array, созданный с помощью кода выше. Напишите код, с помощью которого можно извлечь из него центральный фрагмент размером 3 х 3, с числами 7, 8, 9, 12, 13, 14, 17, 18, 19.

Код должен содержать имя массива и набор индексов для получения нужного среза. В ответе использовать только положительные значения индексов и введите ответ в одну строку без пробелов.

#### Solution:

In [5]:
my_array = np.array([[1,2,3,4,5],
                     [6,7,8,9,10],
                     [11,12,13,14,15],
                     [16,17,18,19,20],
                     [21,22,23,24,25]])

In [6]:
my_array[1:4,1:4] 

array([[ 7,  8,  9],
       [12, 13, 14],
       [17, 18, 19]])

#### Conclusion: the code  above is the right answer!!!

#### Task 5.
Создайте массив my_sin, состоящий из синусов элементов массива my_array. Посчитайте, чему равна сумма элементов полученного массива. Ответ округлите до трёх цифр после запятой.

#### Solution:

In [7]:
my_sin = np.sin(my_array)
my_sin_sum = np.sum(my_sin)
print(f'Сумма элементов полученного массива: {my_sin_sum:.3f}')

Сумма элементов полученного массива: -0.058


#### Conclusion: -0.058 is the right answer!!!

#### Task 6.
Замените элементы в центральном фрагменте 3 х 3 массива my_sin на единицы.Чему равна сумма элементов изменённого массива? Ответ округлите до трёх знаков после запятой.

#### Solution:

In [9]:
my_sin[1:4, 1:4] = 1
my_sin_sum = np.sum(my_sin)
print(f'Сумма элементов полученного массива: {my_sin_sum:.3f}')

Сумма элементов полученного массива: 7.572


#### Conclusion: 7.572 is the right answer!!!

#### Task 7.
Преобразуйте первые 4 столбца массива my_sin в массив из 10 строк и 2 столбцов. Чему равна сумма элементов первого столбца получившегося массива? Ответ округлите до трёх заков после запятой.

#### Solution:

In [12]:
my_sin_crop_reshaped = np.reshape(my_sin[:, :4], (10, 2))
my_sin_crop_reshaped_sum = my_sin_crop_reshaped[:, 0].sum()
print(f'Сумма элементов первого столбца: {my_sin_crop_reshaped_sum:.3f}')

Сумма элементов первого столбца: 2.406


#### Conclusion: 2.406 is the riight answer!!!

### Common preparations. Подготовка массива к заданиям 8-10.
Создайте массив bigdata, содержащий квадраты всех нечётных чисел в диапазоне от 100 до 1000.

In [24]:
bigdata = np.array([i ** 2 for i in range(101, 1001, 2)])

#### Task 8.
Чему равна медиана массива bigdata?

#### Solution:

In [25]:
np.median(bigdata)

302501.0

#### Conclusion: 302501 is the right answer!!!

#### Task 9.
Чему равно стандартное отклонение для массива bigdata? Ответ округлите до целых:

#### Solution:

In [26]:
print(f'Стандартное отклонение для массива bigdata: {np.std(bigdata):.0f}')

Стандартное отклонение для массива bigdata: 292095


In [27]:
np.std(bigdata)

292095.0301991155

#### Conclusion: 292095 is the right asnwer!!!

#### Task 10.
Чему равен коэффициент корреляции между элементами массива bigdata с чётными и нечётными индексами? Введите полученный ответ без изменений и округлений.

### Solution:

In [32]:
corr_coef = np.corrcoef(bigdata[::2], bigdata[1::2])[0, 1]
corr_coef

0.999999729628576

#### Conclusion: 0.999999729628576 is the right answer!!!